presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/AddExchanges.java

```diff
@@ -277,9 +277,10 @@ public PlanWithProperties visitGroupId(GroupIdNode node, PreferredProperties pre
             return rebaseAndDeriveProperties(node, child);
         }
 
-        private Function<VariableReferenceExpression, Optional<VariableReferenceExpression>> translateGroupIdVariables(GroupIdNode node)
+        private Function<RowExpression, Optional<RowExpression>> translateGroupIdVariables(GroupIdNode node)
         {
             return variable -> {
+                checkArgument(variable instanceof VariableReferenceExpression, "Expect VariableReferenceExpression");
```

Comment:
Please add the actual type, just in case it ever fails

Reply:
Theoretically I think returning `Optional.empty()` for other cases is right. But we really should not see anything that's not a `VariableReferenceExpression` atm. I think it's safer to have this check for now.

Reply:
It will also work. I don't have a strong opinion here. It just that it looks kinda confusing =)